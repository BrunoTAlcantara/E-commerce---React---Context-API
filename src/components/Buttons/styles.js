import styled, { css } from 'styled-components';
import { ShoppingCart as CartIcon } from '@styled-icons/material-outlined/ShoppingCart';

const buttonSize = {
  normal: () => css`
    width: 15rem;
    height: 4rem;
  `,
  big: () => css`
    width: 35rem;
    height: 4.2rem;
  `,
};
const buttonColor = {
  primary: (theme) => css`
    color: ${theme.colors.secondaryColor};
    background-color: ${theme.colors.primaryColor};
  `,
  secondary: (theme) => css`
    color: ${theme.colors.primaryColor};
    background-color: ${theme.colors.secondaryColor};
  `,
  disabled: (theme) => css`
    color: ${theme.colors.primaryColor};
    border: 0.3rem solid ${theme.colors.primaryColor};
    background: none;
  `,
};

export const Container = styled.button`
  ${({ theme, size, color }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 1rem;
    gap: 1rem;
    border: none;
    border-radius: 0.5rem;
    ${buttonSize[size]}
    ${buttonColor[color](theme)}
  `}
`;

export const Icon = styled(CartIcon)`
  width: 2.5rem;
  right: 2.5rem;
`;
