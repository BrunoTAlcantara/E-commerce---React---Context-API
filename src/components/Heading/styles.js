import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) =>
    css`
      font-size: ${theme.font.sizes.small};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.sizes.medium};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.sizes.big};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.font.sizes.huge};
      ${mediaFont(theme)};
    `,
};
const titleWeight = {
  bold: (theme) =>
    css`
      font-weight: ${theme.font.weight.bold};
    `,
  eBold: (theme) =>
    css`
      font-weight: ${theme.font.weight.eBold};
    `,
  regular: (theme) =>
    css`
      font-weight: ${theme.font.weight.regular};
    `,
  thin: (theme) => css`
    font-weight: ${theme.font.weight.thin};
  `,
};
const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;
const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.big};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, weight, uppercase }) => css`
    color: ${colorDark
      ? theme.colors.primaryColor
      : theme.colors.secondaryColor};
    ${titleCase(uppercase)}
    ${titleSize[size](theme)};
    ${titleWeight[weight](theme)};
  `}
`;
