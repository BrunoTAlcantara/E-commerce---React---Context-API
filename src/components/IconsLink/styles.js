import styled, { css } from 'styled-components';
import { AccountCircle } from '@styled-icons/material-outlined/AccountCircle';
import { ShoppingCart } from '@styled-icons/material-outlined/ShoppingCart';

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
  `}
`;
export const Button = styled.button`
  ${({ theme }) => css`
    width: 3rem;
    height: 3rem;
    cursor: pointer;

    color: ${theme.colors.primaryColor};
    border: none;
    margin-left: 1rem;
    background: none;
    @media ${theme.media.lteMedium} {
      width: 2rem;
      height: 2rem;
      font-size: ${theme.font.sizes.medium};
    }
  `}
`;
