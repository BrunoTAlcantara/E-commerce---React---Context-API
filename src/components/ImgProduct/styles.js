import styled, { css } from 'styled-components';

const imgSize = {
  small: () => css`
    width: 8rem;
    height: 8rem;
  `,
  normal: (theme) => css`
    width: 30rem;
    height: 30rem;
    @media ${theme.media.lteMedium} {
      width: 20rem;
      height: 20rem;
    }
  `,
  big: (theme) => css`
    width: 56rem;
    height: 56rem;
    @media ${theme.media.lteMedium} {
      width: 30rem;
      height: 30rem;
    }
  `,
};

export const Container = styled.img`
  ${({ theme, size }) => css`
    ${imgSize[size](theme)}

    border-radius: 0.5rem;
    border: none;
  `}
`;
