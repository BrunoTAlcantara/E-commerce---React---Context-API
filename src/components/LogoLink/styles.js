import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    text-decoration: none;
    color: inherit;
    font-size: ${theme.font.sizes.big};
    font-weight: ${theme.font.weight.eBold};

    > img {
      max-height: 2.5rem;
    }
    @media ${theme.media.lteMedium} {
      max-height: 2.5rem;
      font-size: ${theme.font.sizes.medium};
    }
  `}
`;
