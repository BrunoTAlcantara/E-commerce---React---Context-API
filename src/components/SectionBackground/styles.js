import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;

    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    align-items: center;
    .boxWrap {
      width: 100%;

      @media (min-width: 1000px) {
        display: flex;
        flex-flow: row nowrap;
        justify-content: center;
        align-content: center;
        align-items: center;
      }
    }
  `}
`;
