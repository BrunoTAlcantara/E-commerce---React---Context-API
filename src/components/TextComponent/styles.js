import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.weight.regular};
  `}
`;
