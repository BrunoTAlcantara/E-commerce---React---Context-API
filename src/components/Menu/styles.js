import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
  display: flex;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    z-index: 5;
    top: 0;
    left: 0;
    width: 100%;

    padding: 1rem 15rem;
    background: ${theme.colors.secondaryColor};
    border-bottom: #ddd;
    ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
    @media ${theme.media.lteMedium} {
      visibility: hidden;
      opacity: 0;
      display: none;
      height: 100%;
      ${visible && menuVisible(theme)}

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100%;
        align-items: center;
        justify-content: center;
      }
    }
  `}
`;
export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    margin: 1rem;
    justify-content: space-between;
    align-items: center;
    background: ${theme.colors.secondaryColor};

    @media ${theme.media.lteMedium} {
      display: block;
      text-align: left;
      padding: ${theme.font.spacing.big} 0;
      > h1 {
        display: none;
      }
      > div {
        display: none;
      }
    }
  `}
`;
export const Button = styled.button`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      right: 2rem;
      width: 3rem;
      height: 3rem;
      background: none;
      color: ${theme.colors.primaryColor};
      border: none;
      margin: 1rem;
    }
  `}
`;
export const MenuMob = styled.div`
  ${({ theme }) => css`
    display: none;
    @media ${theme.media.lteMedium} {
      display: flex;
      justify-content: space-between;
      align-items: center;
      z-index: 3;
      top: 0;
      left: 0;
      right: 0;
      width: 100%;
      padding: 1rem;

      background: ${theme.colors.secondaryColor};
      border-bottom: #ddd;
    }
  `}
`;
