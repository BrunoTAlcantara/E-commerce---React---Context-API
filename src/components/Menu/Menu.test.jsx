import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Menu } from '.';
import { theme } from '../../styles/theme';
import linksMock from '../NavLinks/mock';

const logoData = {
  text: 'Logo',
  link: '#target',
};

describe('<Menu />', () => {
  it('should render Logo and Main menu Nav', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );
    expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'Main Menu' }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  it('should  render menu mobile and button for open e close menu', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );
    const button = screen.getByLabelText('Open/Close Menu');
    const menuContainer = button.nextSibling;

    expect(button).toHaveStyleRule('display', 'none');
    expect(button).toHaveStyleRule('display', 'block', {
      media: theme.media.lteMedium,
    });
    expect(screen.getByLabelText('Open Menu')).toBeInTheDocument();

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });
    fireEvent.click(button);
    expect(screen.getByLabelText('Close Menu')).toBeInTheDocument();
    expect(menuContainer).toHaveStyleRule('opacity', '1', {
      media: theme.media.lteMedium,
    });

    fireEvent.click(button);
    expect(screen.getByLabelText('Open Menu')).toBeInTheDocument();

    expect(menuContainer).toHaveStyleRule('opacity', '0', {
      media: theme.media.lteMedium,
    });
  });
});
