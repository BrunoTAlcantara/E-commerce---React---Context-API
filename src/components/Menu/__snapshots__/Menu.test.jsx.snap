// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render Logo and Main menu Nav 1`] = `
.c9 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 2.4rem;
}

.c7 {
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  padding: 1rem;
  background: #E5E5E5;
  border-bottom: #ddd;
}

.c7 .c8 {
  padding-top: 0;
  padding-bottom: 0;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 1rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #E5E5E5;
}

.c0 {
  display: none;
}

.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  width: 3rem;
  height: 3rem;
  cursor: pointer;
  color: #252A2C;
  border: none;
  margin-left: 1rem;
  background: none;
}

.c4 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-size: 2.4rem;
  font-weight: 800;
}

.c4 > img {
  max-height: 2.5rem;
}

.c3 {
  color: #252A2C;
  text-transform: none;
  font-size: 2.4rem;
  font-weight: 700;
}

.c12 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  font-weight: 700;
  color: #252A2C;
  padding: 0.8rem;
  position: relative;
}

.c12::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #252A2C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c12:hover::after {
  left: 25%;
  width: 50%;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

@media (max-width:768px) {
  .c7 {
    visibility: hidden;
    opacity: 0;
    display: none;
    height: 100%;
  }

  .c7 > .c8 {
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100%;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
  }
}

@media (max-width:768px) {
  .c10 {
    display: block;
    text-align: left;
    padding: 2.4rem 0;
  }

  .c10 > h1 {
    display: none;
  }

  .c10 > div {
    display: none;
  }
}

@media (max-width:768px) {
  .c1 {
    right: 2rem;
    width: 3rem;
    height: 3rem;
    background: none;
    color: #252A2C;
    border: none;
    margin: 1rem;
  }
}

@media (max-width:768px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    z-index: 3;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    padding: 1rem;
    background: #E5E5E5;
    border-bottom: #ddd;
  }
}

@media (max-width:768px) {
  .c6 {
    width: 2rem;
    height: 2rem;
    font-size: 1.6rem;
  }
}

@media (max-width:768px) {
  .c4 {
    max-height: 2.5rem;
    font-size: 2.4rem;
  }
}

@media (max-width:768px) {
  .c11 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<div>
  <div
    class="c0"
  >
    <button
      aria-label="Open/Close Menu"
      class="c1"
    >
      <svg
        aria-hidden="true"
        aria-label="Open Menu"
        class="c2"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <h1
      class="c3"
    >
      <a
        class="c4"
        href="#target"
      >
        Logo
      </a>
    </h1>
    <div
      class="c5"
    >
      <button
        aria-label="Button User"
        class="c6"
        href="google.com"
      >
        <svg
          aria-hidden="true"
          class="c2"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
          />
        </svg>
      </button>
      <button
        class="c6"
      >
        <svg
          aria-hidden="true"
          aria-label="Button Cart"
          class="c2"
          fill="currentColor"
          focusable="false"
          href="google.com.br"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    class="c7"
  >
    <div
      class="c8 c9"
    >
      <h1>
        <div
          class="c10"
        >
          <h1
            class="c3"
          >
            <a
              class="c4"
              href="#target"
            >
              Logo
            </a>
          </h1>
          <nav
            aria-label="Main Menu"
            class="c11"
          >
            <a
              class="c12"
              href="#target1"
              target="_self"
            >
              link1
            </a>
            <a
              class="c12"
              href="#target2"
              target="_self"
            >
              link2
            </a>
            <a
              class="c12"
              href="#target3"
              target="_self"
            >
              link3
            </a>
            <a
              class="c12"
              href="#target4"
              target="_self"
            >
              link4
            </a>
          </nav>
          <div
            class="c5"
          >
            <button
              aria-label="Button User"
              class="c6"
              href="google.com"
            >
              <svg
                aria-hidden="true"
                class="c2"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zM7.07 18.28c.43-.9 3.05-1.78 4.93-1.78s4.51.88 4.93 1.78C15.57 19.36 13.86 20 12 20s-3.57-.64-4.93-1.72zm11.29-1.45c-1.43-1.74-4.9-2.33-6.36-2.33s-4.93.59-6.36 2.33A7.95 7.95 0 014 12c0-4.41 3.59-8 8-8s8 3.59 8 8c0 1.82-.62 3.49-1.64 4.83zM12 6c-1.94 0-3.5 1.56-3.5 3.5S10.06 13 12 13s3.5-1.56 3.5-3.5S13.94 6 12 6zm0 5c-.83 0-1.5-.67-1.5-1.5S11.17 8 12 8s1.5.67 1.5 1.5S12.83 11 12 11z"
                />
              </svg>
            </button>
            <button
              class="c6"
            >
              <svg
                aria-hidden="true"
                aria-label="Button Cart"
                class="c2"
                fill="currentColor"
                focusable="false"
                href="google.com.br"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M15.55 13c.75 0 1.41-.41 1.75-1.03l3.58-6.49A.996.996 0 0020.01 4H5.21l-.94-2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2h7.45zM6.16 6h12.15l-2.76 5H8.53L6.16 6zM7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
            </button>
          </div>
        </div>
      </h1>
    </div>
  </div>
</div>
`;
